examples/mqtt/aws_iot/simulated_home_auto_gen_topic.py

```diff
@@ -0,0 +1,209 @@
+#
+# A Simulated DHT sensor with Temperature and Humidity Metrics and a Simulated
+# Digital Light sensor with binary luminance Metrics are used.
+#
+#
+#                                            Dell5kEdgeSystem
+#                                                   |
+#                                                   |
+#                                                   |
+#                     -----------------------------------------------------
+#                    |                              |                      |
+#                    |                              |                      |
+#                    |                              |                      |
+#                 DHT Sensor                 Digital Light Sensor    CPU Utilization
+#                    |                              |                   Metric
+#                    |                              |
+#            ----------------                  Light Metric
+#           |                |
+#           |                |
+#      Temperature        Humidity
+#         Metric           Metric
+#
+# Developers can leverage the following options:
+# ---------------------------------------------
+#  a) Use single publish and subscribe topic generated by LIOTA for an EdgeSystem, its Devices and its Metrics.
+#  b) Use custom single publish and subscribe topic for an EdgeSystem, its Devices and Metrics.
+#
+#  - In the above two cases, MQTT message's payload MUST be self-descriptive so that subscriber can subscribe
+#    process accordingly to a single topic by parsing payload.
+#
+#  c) Use custom publish and subscribe topics for Metrics.
+#  - In this case, MQTT message's payload need not be self-descriptive.  Subscribers can subscribe to
+#    appropriate topics and process accordingly.
+#
+#  d) Use combination of (a) and (c) or (b) and (c).
+#
+#
+# GenericMqtt DCC has enclose_metadata option.  It can be used to enclose EdgeSystem, Device and Metric names
+# along with the sensor data payload of a Metric.
+#
+# This example showcases publishing Metrics using (a) and enclose_metadata
+# -----------------------------------------------------------------------------------------------------------------
+
+if __name__ == '__main__':
+    #  Creating EdgeSystem
+    edge_system = Dell5KEdgeSystem(config['EdgeSystemName'])
+    # Encapsulate TLS parameters
+    tls_conf = MqttTLSConf(config['cert_required'], config['tls_version'], config['cipher'])
+    # Encapsulate QoS related parameters
+    qos_details = QoSDetails(config['in_flight'], config['queue_size'], config['retry'])
+
+    #  Connecting to AWSIoT
+    #  Initializing GenericMqtt DCC using MqttDccComms
+    #  AWSIoT broker doesn't support session persistence.  So, always use "clean_session=True"
+    #  Publish topic for all Metrics will be 'liota/generated_local_uuid_of_edge_system'
+    aws = GenericMqtt(MqttDccComms(edge_system=edge_system, url=config['BrokerIP'], port=config['BrokerPort'],
```

Comment:
Edge_system to hold name as well as client certificates, don't take them individually in any DCC_comms.